cocotb/monitors/xgmii.py

```diff
@@ -133,6 +133,14 @@ def _monitor_recv(self):
                     yield clk
                     ctrl, bytes = self._get_bytes()
 
+            if ctrl[4] and bytes[4] == _XGMII_START:
```

Comment:
Won't this give an `IndexError` if `self.bytes == 4`?

Should this be an `elif`?

Why only bytes 0 and 4? Can you point to some documentation that says in which positions `START` is allowed to appear?

Reply:
Is `START` allowed to appear in the same word as `_XGMII_TERMINATE`?

Reply:
Start and Terminate cannot appear on the same word as the preamble is 8bytes and even with 1 byte of data it would be on the subsequent word.

https://www.xilinx.com/support/documentation/ip_documentation/ten_gig_eth_mac/v14_0/pg072-ten-gig-eth-mac.pdf
Page 78-79 for more information on offsets within XGMII, but basically in 10G MACs (using) there is a interframe gap of 12 bytes so if you have a 8 byte word you will invariable have a offset word.

You could be correct, there may be a issue with a 4 byte xgmii interface, as mine is 8 bytes